modules/worker-utils/src/lib/worker-farm/worker-body.ts

```diff
@@ -31,24 +40,26 @@ export default class WorkerBody {
   static set onmessage(onMessage: (type: WorkerMessageType, payload: WorkerMessagePayload) => any) {
     function handleMessage(message) {
       // Confusingly the message itself also has a 'type' field which is always set to 'message'
-      const parentPort = getParentPort();
-      const {type, payload} = parentPort ? message : message.data;
-      // if (!isKnownMessage(message)) {
-      //   return;
-      // }
-      onMessage(type, payload);
+      getParentPort().then((parentPort) => {
```

Comment:
We can't use async/await syntax? Maybe add a comment explaining why we are using `.then()`?

Reply:
`await` is not allowed for setter but I will use it for methods